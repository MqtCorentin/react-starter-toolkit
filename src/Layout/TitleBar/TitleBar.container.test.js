import React from 'react';
import { StaticRouter } from 'react-router-dom/server';
import { render } from '@testing-library/react';
import TitleBarEnhanced from './index';

const wrapper = ({ children }) => <StaticRouter context={{}}>{children}</StaticRouter>;

const defaultProps = { backHome: false, classModifier: '' };

const createRender = (customProps = {}) => {
  const actualProps = { ...defaultProps, ...customProps };
  return render(<TitleBarEnhanced {...actualProps} />, { wrapper });
};

describe('Component <TitleBarEnhanced />', () => {
  it('Should render TitleBarEnhanced', () => {
    const { asFragment } = createRender();
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should className contain "af-title-bar--red" when classModifier equal "red"', () => {
    const { container, asFragment } = createRender({ classModifier: 'red' });
    expect(container.querySelector('.af-title-bar--red')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should className contain "af-title-bar--backhome" modifier when backhome equal true', () => {
    const { container, asFragment } = createRender({ backHome: true });
    expect(container.querySelector('.af-title-bar--backhome')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
