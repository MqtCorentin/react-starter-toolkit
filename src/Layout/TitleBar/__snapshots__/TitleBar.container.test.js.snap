// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <TitleBarEnhanced /> Should render TitleBarEnhanced 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar af-title-bar--red"
  >
    <div
      class="container-fluid container af-title-bar__wrapper"
    >
      <h1
        class="af-title-bar__title"
      />
    </div>
  </div>
</DocumentFragment>
`;
